File: mobile/src/utils/dateRange.js
// Port dari frontend/src/pages/SchedulePage.jsx.
export const toYyyyMm = (d) => {
  const yyyy = d.getFullYear()
  const mm = String(d.getMonth() + 1).padStart(2, '0')
  return `${yyyy}-${mm}`
}

export const getMonthRange = (yyyyMm) => {
  const [y, m] = String(yyyyMm).split('-').map((v) => parseInt(v, 10))
  const start = new Date(y, (m || 1) - 1, 1)
  const end = new Date(y, m || 1, 0)
  const toIsoDate = (dt) => dt.toISOString().slice(0, 10)
  return { startDate: toIsoDate(start), endDate: toIsoDate(end) }
}

File: mobile/lib/utils/date_range.dart
// Port dari frontend/src/pages/SchedulePage.jsx.
String toYyyyMm(DateTime d) {
  final mm = d.month.toString().padLeft(2, '0');
  return '${d.year}-$mm';
}

class MonthRange {
  final String startDate;
  final String endDate;
  MonthRange(this.startDate, this.endDate);
}

MonthRange getMonthRange(String yyyyMm) {
  final parts = yyyyMm.split('-');
  final y = int.parse(parts[0]);
  final m = int.parse(parts[1]);
  final start = DateTime(y, m, 1);
  final end = DateTime(y, m + 1, 0);
  String toIsoDate(DateTime dt) =>
      '${dt.year.toString().padLeft(4, '0')}-${dt.month.toString().padLeft(2, '0')}-${dt.day.toString().padLeft(2, '0')}';
  return MonthRange(toIsoDate(start), toIsoDate(end));
}
